### PR TITLE
UAR-293: Check your answers - New BO

### PR DESCRIPTION
**Pull request by jkayyala**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-293

### Change description
Add changes for showing Beneficial Owners in Check your Answers

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/update/update-check-your-answers.html [hunk 1]
```diff
@@ -18,6 +18,20 @@ <h1 class="govuk-heading-xl">Check your answers before sending your application
       {% include "includes/check-your-answers/verification-checks.html" %}
       <br>
       {% include "includes/entity.html" %}
+      {% if (appData.beneficial_owners_individual and appData.beneficial_owners_individual.length > 0)
+        or (appData.beneficial_owners_corporate and appData.beneficial_owners_corporate.length > 0)
+        or (appData.beneficial_owners_government_or_public_authority and appData.beneficial_owners_government_or_public_authority.length > 0) %}
+        <br><h2 class="govuk-heading-l">Beneficial owner details</h2>
```
Comment (AndreKramer7654) [inline views/update/update-check-your-answers.html:24]: Don't you need to filter by ch_reference not being undefined for new BOs?

Comment (AndreKramer7654) [reply views/update/update-check-your-answers.html:24]: Assuming we are showing the new BOs separate from reviewed BOs.

File: views/includes/check-your-answers/beneficial-owner-other.html [hunk 2]
```diff
@@ -53,6 +53,12 @@
   {% endif %}
 {% endset %}
 
+{% if pageParams.isRegistration %}
+  {% set changeLinkUrl = OE_CONFIGS.BENEFICIAL_OWNER_OTHER_URL %}
+{% else %}
+  {% set changeLinkUrl = OE_CONFIGS.UPDATE_BENEFICIAL_OWNER_OTHER_URL %}
+{% endif %}
+
```
Comment (azafarCH) [inline views/includes/check-your-answers/beneficial-owner-other.html:61]: extra line here but not one before heading in beneficial-owner-gov template

File: test/controllers/check.your.answers.controller.spec.ts [hunk 3]
```diff
@@ -65,6 +65,9 @@ import {
   SERVICE_UNAVAILABLE,
   SOMEONE_ELSE_REGISTERING,
   TRUST_INFORMATION_LINK,
+  CHANGE_LINK_BO_OTHER,
+  CHANGE_LINK_BO_GOV,
+  CHANGE_LINK_BO_IND,
```
Comment (sjahir) [inline test/controllers/check.your.answers.controller.spec.ts:70]: remove comma

File: test/__mocks__/text.mock.ts [hunk 4]
```diff
@@ -158,6 +158,9 @@ export const CHANGE_LINK_INDIVIDUAL_BO_SERVICE_ADDRESS = CHANGE_LINK_INDIVIDUAL_
 export const CHANGE_LINK_INDIVIDUAL_BO_START_DATE = CHANGE_LINK_INDIVIDUAL_BO + "Ivan Drago - date they became a beneficial owner";
 export const CHANGE_LINK_INDIVIDUAL_BO_NOC = CHANGE_LINK_INDIVIDUAL_BO + "Ivan Drago - nature of control";
 export const CHANGE_LINK_INDIVIDUAL_BO_IS_ON_SANCTIONS_LIST = CHANGE_LINK_INDIVIDUAL_BO + "Ivan Drago - are they on the sanctions list?";
+export const CHANGE_LINK_BO_IND = "/register-an-overseas-entity/beneficial-owner-individual/7ab403a0-4459-4ac0-b500-6d6f314da5b2";
```
Comment (sjahir) [inline test/__mocks__/text.mock.ts:161]: would CHANGE_LINK_BO_INDIVIDUAL read better?

File: test/__mocks__/text.mock.ts [hunk 5]
```diff
@@ -176,6 +179,9 @@ export const CHANGE_LINK_ENTITY_SERVICE_ADDRESS = "/update-an-overseas-entity/en
 export const CHANGE_LINK_ENTITY_EMAIL = "/update-an-overseas-entity/entity#email";
 export const CHANGE_LINK_ENTITY_LEGAL_FORM = "/update-an-overseas-entity/entity#legal_form";
 export const CHANGE_LINK_ENTITY_GOVERNING_LAW = "/update-an-overseas-entity/entity#law_governed";
+export const UPDATE_CHANGE_LINK_BO_IND = "/update-an-overseas-entity/update-beneficial-owner-individual/7ab403a0-4459-4ac0-b500-6d6f314da5b2";
```
Comment (sjahir) [inline test/__mocks__/text.mock.ts:182]: same here as well

File: views/update/update-check-your-answers.html [hunk 6]
```diff
@@ -18,6 +18,25 @@ <h1 class="govuk-heading-xl">Check your answers before sending your application
       {% include "includes/check-your-answers/verification-checks.html" %}
       <br>
       {% include "includes/entity.html" %}
+      <br><h2 class="govuk-heading-l">Beneficial owner details</h2>
+        {% if (appData.beneficial_owners_individual and appData.beneficial_owners_individual.length > 0) %}
+          {% set savedIndBO = appData.beneficial_owners_individual | rejectattr("ch_reference", "undefined") %}
```
Comment (sjahir) [inline views/update/update-check-your-answers.html:23]: consider changing to savedIndividualBo

**Review by AndreKramer7654 — APPROVED** [review]

**Review by azafarCH — APPROVED** [review]

**Review by sjahir — APPROVED** [review]

**Review by bantao-ch — APPROVED** [review]

**Pull request merged**